src/analysis/lattice.h

```diff
@@ -42,15 +43,31 @@ constexpr bool is_lattice = has_getBottom<Lattice>&& has_compare<Lattice>&&
 // Represents a powerset lattice which is constructed from a finite set which
 // can be represented by a bitvector. Set elements are represented by
 // FinitePowersetLattice::Element, which represents members present in each
-// element by bits in the bitvector.
-class FinitePowersetLattice {
+// element by bits in the bitvector. Type is the type of a member in a lattice
+// element.
+template<typename Type> class FinitePowersetLattice {
```

Comment:
`Type` is a very important class in general in our codebase, so overloading the term here can be a little confusing I think. Perhaps just `T`? That is what we use for similar templating in `SmallVector` and `SmallSet` for example.